### ISSUE TITLE
MergeFromTo - TypeError: Cannot read property 'failed' of null

### ISSUE THREAD

**Issue by timdyck** [steveukx/git-js#227]
Hey!

I am using the `mergeFromTo` method, and I have been working on handling merge conflicts.

I forced a conflict by creating a change locally, then committing a conflict on remote.  Then, using simple-git, I fetch the remote and attempt to merge `origin/master` to `master`.

```
[2017-12-12T00:47:43.461Z]  INFO: IL-log/26854 on tdyck-mbp: Fetched (repository={{...}})
error: Your local changes to the following files would be overwritten by merge:
	{{...}}
Please commit your changes or stash them before you merge.
Aborting

[2017-12-12T00:47:43.508Z] FATAL: IL-log/26854 on tdyck-mbp: Uncaught exception (err.code="")
    TypeError: Cannot read property 'failed' of null
        at mergeHandler (/Users/tdyck/code/web/node_modules/simple-git/src/git.js:841:26)
        at Git.<anonymous> (/Users/tdyck/code/web/node_modules/simple-git/src/git.js:1443:13)
        at Function.Git.fail (/Users/tdyck/code/web/node_modules/simple-git/src/git.js:1368:18)
        at Git.<anonymous> (/Users/tdyck/code/web/node_modules/simple-git/src/git.js:1333:20)
        at emitTwo (events.js:106:13)
        at ChildProcess.emit (events.js:194:7)
        at maybeClose (internal/child_process.js:899:16)
        at Socket.<anonymous> (internal/child_process.js:342:11)
        at emitOne (events.js:96:13)
        at Socket.emit (events.js:191:7)
        at Pipe._handle.close [as _onclose] (net.js:511:12)
```

The fetch result is expected, but the following error seems internal to simple-git.  My call to `mergeFromTo` looks like the following:

```
var GIT_SSH_COMMAND = "ssh -o StrictHostKeyChecking=no -i " + GITHUB_PRIV_KEY_PATH;
git(this.repoPath)
            .env('GIT_SSH_COMMAND', GIT_SSH_COMMAND)
            .mergeFromTo(from, to, function (err, data) {...}
```

I do an `if (!err) ...` check, and attempt to handle the error, but the promise never completes because of the `Cannot read property 'failed' of null` error above.

Let me know if you need any more information, or if you think I am doing something incorrect on my end.

Tim

**Comment by steveukx** [comment]
Thank you for the full stack trace... the `merge` handler wasn't handling the case where there was a non-clean shutdown in the `git merge` command. This is resolved now and the new version `1.86.0` is on npm.

**Comment by timdyck** [comment]
Thank you!
